kiota_http/middleware/parameters_name_decoding_handler.py

```diff
@@ -67,30 +69,33 @@ def _get_current_options(self, request: httpx.Request) -> ParametersNameDecoding
         Returns:
             ParametersNameDecodingHandlerOption: The options to used.
         """
-        if options := getattr(request, 'options', None):
+        if options := getattr(request, "options", None):
             current_options = options.get(  # type:ignore
                 ParametersNameDecodingHandlerOption.get_key(), self.options
             )
             return current_options
         return self.options
 
-    @staticmethod
-    def decode_uri_encoded_string(original: str, characters_to_decode: List[str]) -> str:
-        """Decodes a uri encoded string ."""
+    def decode_uri_encoded_string(self, original: str, characters_to_decode: List[str]) -> str:
+        """Decodes a uri encoded url string"""
         if not original or not characters_to_decode:
             return original
+
         query_params = urlparse(original).query
         if not query_params:
-            return original
+            return unquote(original)
```

Comment:
why do we need to unquote the original if we don't have any query parameter?

Reply:
This i was unsure about, before the change to only decode the query param values, the function decoded the entire string (including the base of it) so i wasn;t sure if we wanted to bing that functionality back.

Reply:
it was an implementation drift compared with the other languages. The sole responsibility of this handler is to decode parameter names. We're doing that because some characters are reserved as URI template parameter names according to RFC 6570 so we had to encode them when projecting the code.